server/server.go

```diff
@@ -52,7 +55,22 @@ func Listen(cfg config.Configuration, handler http.Handler, metrics *metrics.Con
 	// Then block the thread. When the OS sends a shutdown signal, alert each of the servers.
 	// Once they're finished shutting down (the "done" channel gets pinged for each server),
 	// this funciton can return.
-	wait(stopSignals, done, stopMain, stopAdmin)
+	if cfg.Metrics.Prometheus.Port != 0 {
```

Comment:
This should only need to check `cfg.Metrics.Enabled` since it will be forced to false by the validator if the port is 0. If the prometheus metrics are disabled this code will currently fail on a nil registry.

Reply:
Done